sdk-java/src/main/java/com/spotify/confidence/telemetry/Telemetry.java

```diff
@@ -0,0 +1,53 @@
+package com.spotify.confidence.telemetry;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.spotify.telemetry.v1.LibraryTraces;
+import com.spotify.telemetry.v1.Monitoring;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Telemetry {
+  private final ConcurrentLinkedQueue<LibraryTraces.Trace> latencyTraces =
+      new ConcurrentLinkedQueue<>();
+  private final AtomicBoolean isProvider = new AtomicBoolean(false);
+
+  public void appendLatency(long latency) {
+    latencyTraces.add(
+        LibraryTraces.Trace.newBuilder()
+            .setId(LibraryTraces.TraceId.TRACE_ID_RESOLVE_LATENCY)
+            .setMillisecondDuration(latency)
+            .build());
+  }
+
+  public Monitoring getSnapshot() {
+    final Monitoring snapshot = getSnapshotInternal();
+    clear();
```

Comment:
Could there be a risk that data is added to `latencyTracers` in between these two lines, and get lost? Perhaps making all these methods synchronized can ensure batter safety

Reply:
Yes there is a risk. But I would argue that this risk is acceptable. It's a bit like sampling 😸